skyportal/handlers/api/shift.py

```diff
@@ -136,32 +136,64 @@ def get(self, group_id=None):
                 schema: Error
         """
         if group_id is not None:
-            shifts = (
+            queried_shifts = (
                 Shift.query_records_accessible_by(
                     self.current_user,
                     mode="read",
                     options=[
-                        joinedload(Shift.group).joinedload(Group.users),
-                        joinedload(Shift.users),
+                        joinedload(Shift.group).joinedload(Group.group_users),
+                        joinedload(Shift.shift_users),
                     ],
                 )
                 .filter(Shift.group_id == group_id)
                 .order_by(Shift.start_date.asc())
                 .all()
             )
         else:
-            shifts = (
+            queried_shifts = (
                 Shift.query_records_accessible_by(
                     self.current_user,
                     mode="read",
                     options=[
-                        joinedload(Shift.group).joinedload(Group.users),
-                        joinedload(Shift.users),
+                        joinedload(Shift.group).joinedload(Group.group_users),
+                        joinedload(Shift.shift_users),
                     ],
                 )
                 .order_by(Shift.start_date.asc())
                 .all()
             )
+        shifts = []
+        for shift in queried_shifts:
+            susers = [
+                {
+                    "id": su.user.id,
+                    "username": su.user.username,
+                    "first_name": su.user.first_name,
+                    "last_name": su.user.last_name,
+                    "contact_email": su.user.contact_email,
+                    "contact_phone": su.user.contact_phone,
+                    "admin": su.admin,
+                }
+                for su in shift.shift_users
+            ]
+            gusers = [
+                {
+                    "id": gu.user.id,
+                    "username": gu.user.username,
+                    "first_name": gu.user.first_name,
+                    "last_name": gu.user.last_name,
+                    "contact_email": gu.user.contact_email,
+                    "contact_phone": gu.user.contact_phone,
+                    "admin": gu.admin,
+                }
```

Comment:
Did you try using `user.to_dict()`? Here and above.

Reply:
Hey @guynir42 ! Thanks for your review. Few questions: What do you mean exactly ? Using user.to_dict() rather than creating a new dict ? I can try of course. The idea here was to be able to create a new user object with only the info we need for shift users, so their name, username, id, contact and if they are an admin or not. If i only use user.to_dict(), i also need to add the admin boolean to it right ? (that is contained in the shift_user, that itself contains user)

Reply:
I think each object that comes from baselayer (or from sql alchemy, I'm not sure) has this `to_dict()` method. It just creates a dictionary from the properties of the ORM object that have been loaded. It will just save you a lot of trouble copying many properties manually. You can then delete or add some keys to fit exactly what you want. 
Make sure to verify you are getting all the properties you expect because some things aren't loaded by default from the database row and so they don't get into the dictionary automatically.

Reply:
Great ! I'll try that and then ask for your opinion ! Thanks again.

Reply:
Done 🚀